lxd/patches.go

```diff
@@ -181,6 +182,51 @@ func patchesApply(d *Daemon, stage patchStage) error {
 
 // Patches begin here
 
+// patchDBNodesAutoInc re-creates the nodes table id column as AUTOINCREMENT.
+// Its done as a patch rather than a schema update so we can use PRAGMA foreign_keys = OFF without a transaction.
+func patchDBNodesAutoInc(name string, d *Daemon) error {
+	row := d.State().Cluster.DB().QueryRow("SELECT sql FROM sqlite_master WHERE name = 'nodes'")
+
+	var schemaSQL string
+	err := row.Scan(&schemaSQL)
+	if err != nil {
+		return err
+	}
+
+	if strings.Contains(schemaSQL, "id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL") {
```

Comment:
@stgraber not sure if this is what you had in mind to only run on leader, but this should prevent it running multiple times.

Reply:
That's a good protection to have.

My thought was to have call `node.ClusterAddress(d.db)` combined with `d.gateway.LeaderAddress()` and if they're not equal, then skip as we're not on the leader.